compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/GetTextFieldPointerModifier.common.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.text
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.text.selection.TextFieldSelectionManager
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.input.pointer.pointerHoverIcon
+import androidx.compose.ui.text.input.OffsetMapping
+
+internal expect fun getTextFieldPointerModifier(
```

Comment:
I'd make this it as regular modifier (`Modifier.` extension)

Reply:
This code is written in commonMain and we want to make minimal changes in AOSP code base.

Pay attention to this line of code:
`val pointerModifier = getTextFieldPointerModifier(...)`
we don't changed type of `pointerModifier` variable.

Reply:
I mean creating modifier via "get" function looks weird. It's not about increase "minimal changes in AOSP code base" or changing return type

What I suggest is something like:
```kt
val pointerModifier = Modifier.textFieldPointerModifier(...)
```

Reply:
I think - better not chage the symantics of AOSP code.
We just extracted it as is.

Reply:
@igordmn, could you please give your opinion on this?

Reply:
A more ideomatic way is:
```
internal expect fun Modifier.textFieldPointer(): Modifier = if (enabled) {
   ...
   if (isInTouchMode) {
      ...
      this
            .then(tapHandlerModifier)
            .then(selectionModifier)
            .pointerHoverIcon(textPointerIcon)
   }
} else {
   this
}
```

it is the same as `getTextFieldPointerModifier` in terms of semantics

Reply:
done (I guess)